src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -222,7 +212,7 @@ private bool ReportBadTargetType(SyntaxNode syntax, TypeSymbol type, DiagnosticB
                 case TypeKind.Enum:
                 case TypeKind.Delegate:
                 case TypeKind.Interface:
```

Comment:
@gafter delegate creation argument are bound differently. To support that we should postpone argument binding or rebind. Note: postponing might result in missing diagnostics because we may not ever have chance to try to create a conversion.